src/api/structure/testing-unit.js

```diff
@@ -48,11 +49,12 @@ export default class TestingUnit {
 
         assertType(is.string, 'page', 'The page URL', this.pageUrl);
 
-        if (!PROTOCOL_RE.test(this.pageUrl) && this.pageUrl !== 'about:blank') {
-            var protocol = IMPLICIT_PROTOCOL_RE.test(this.pageUrl) ? 'http:' : 'http://';
+        var urlValidation = validateUrl(this.pageUrl);
 
-            this.pageUrl = protocol + this.pageUrl;
-        }
+        if (!urlValidation.valid && urlValidation.reason.protocol)
```

Comment:
And also that's fine to use runtime APIError in action validation if it happens only on server. We can avoid using `test-run` errors if error happens only on server and we don't need to serialize it back and forth. Actually, whole error reporting system became a bit messed up and requires simplification.